### PR TITLE
Search as I move the map

### PR DESCRIPTION
**Pull request by AkkarinForest**
Search as I move checkbox, it's turned off by default. When the map is moved on zoomed it changes into "search again" button. When the sessions list is refreshed it changes back to the checkbox.
Needs styling: https://llp.kanbanery.com/projects/8040/board/tasks/2407695

### CODE REVIEW

File: app/javascript/elm/src/Main.elm [hunk 1]
```diff
@@ -59,6 +59,8 @@ type alias Model =
     , resetIcon : String
     , heatMapThresholds : WebData HeatMapThresholds
     , isStreaming : Bool
+    , isSearchOn : Bool
```
Comment (3v0k4) [inline app/javascript/elm/src/Main.elm]: Have you considered using the full feature name? That means `isSearchAsIMoveOn`. It's a bit confusing with just `isSearchOn`

File: app/javascript/elm/tests/MainTests.elm [hunk 2]
```diff
@@ -984,4 +992,11 @@ updateTests =
                     |> update (UpdateHeatMapMaximum int)
                     |> Tuple.first
                     |> Expect.equal expected
+        , fuzz bool "ToggleIsSearchOn changes the state of model.isSearchOn" <|
```
Comment (3v0k4) [inline app/javascript/elm/tests/MainTests.elm]: Have you considered not testing at all? Also, I see that you haven't tested `update MapMoved`. Any reasons for that?

File: app/javascript/angular/code/controllers/_sessions_list_ctrl.js [hunk 3]
```diff
@@ -48,12 +48,24 @@ export const SessionsListCtrl = (
     ({ hasChangedProgrammatically }) => {
       console.log("watch - params.get('map')");
       if (sessions.hasSelectedSessions()) return;
-      if ($scope.firstLoad || hasChangedProgrammatically) {
-        sessions.fetch({ amount: params.get("fetchedSessionsCount") });
-      } else {
-        sessions.fetch();
+      if (hasChangedProgrammatically === undefined) return;
```
Comment (3v0k4) [inline app/javascript/angular/code/controllers/_sessions_list_ctrl.js]: Let's have a chat about this. Not sure I follow what's going on

File: app/javascript/elm/tests/MainTests.elm (hunk 2 above)
Comment (AkkarinForest) [reply app/javascript/elm/tests/MainTests.elm]: It just feels wrong - not testing at all. I know we discussed overtesting, but I feel uncomfortable not adding any tests when I add new logic. Probably because a few weeks ago when we were adding staff in angular you always suggested that I should do more testing 😄

I understand why it seems like the tests are unnecessary right now. But do you remember how it was joining aircasitng when frontend didn't have any testing at all? 
And if we don't add this test cases now, then no one is gonna add them any time soon, so if the project will grow in a few years it might get to a similar situation as before?

TLDR - I'm experimenting and trying to find a balance 😉

File: app/javascript/elm/src/Main.elm (hunk 1 above)
Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: or `isSearchAsIMove`

File: app/javascript/elm/tests/MainTests.elm (hunk 2 above)
Comment (3v0k4) [reply app/javascript/elm/tests/MainTests.elm]: I think the Angular case is a totally different one (Ruby and Rails too for that matter). TLDR: no type system and no managed runtime like in Elm.

I'm fine with having different opinions on this. As long as we can back them up with valid arguments ;) Let's just test mindfully and not by trying to cover everything possible
